cli/dts/lib.deno.shared_globals.d.ts

```diff
@@ -698,51 +688,37 @@ interface WorkerOptions {
    *     const fileContents = new TextDecoder().decode(buf);
    *     console.log(fileContents);
    * }
-   * ```
-   *
-   * // log.txt
-   * hello world
-   * hello world 2
    *
-   * // run program
-   * $ deno run --allow-read mod.ts
-   * hello world
-   * hello world2
+   * // $ cat log.txt
+   * // hello world
+   * // hello world 2
    *
+   * // $ deno run --allow-read mod.ts
+   * // hello world
+   * // hello world2
+   * ```
    */
   // TODO(Soremwar)
-  // `deno: true` is kept for backwards compatibility with the previous worker
-  // options implementation. Remove for 2.0.
-  deno?: true | {
-    namespace?: boolean;
-    /** Set to `"none"` to disable all the permissions in the worker. */
-    permissions?: "inherit" | "none" | {
-      env?: "inherit" | boolean;
-      hrtime?: "inherit" | boolean;
-      /** The format of the net access list must be `hostname[:port]`
-       * in order to be resolved.
-       *
-       * ```
-       * net: ["https://deno.land", "localhost:8080"],
-       * ```
-       * */
-      net?: "inherit" | boolean | string[];
-      plugin?: "inherit" | boolean;
-      read?: "inherit" | boolean | Array<string | URL>;
-      run?: "inherit" | boolean;
-      write?: "inherit" | boolean | Array<string | URL>;
-    };
-  };
+  // `deno: boolean` is kept for backwards compatibility with the previous
+  // worker options implementation. Remove for 2.0.
+  deno?: boolean;
 }
 
-declare class Worker extends EventTarget {
+// TODO(nayeemrmn): `class Worker` had to be split into `var Worker` +
+// `interface WorkerConstructor` + `interfaceWorker`, so that a constructor
+// overload could be added in `lib.deno.unstable.d.ts`. Remove on stabilization.
+declare var Worker: WorkerConstructor;
+declare interface WorkerConstructor {
```

Comment:
@bartlomieju It ended being quite complex because TS can't "merge" classes from two declaration files.

However, if we're willing to compromise and remove `WorkerOptions::deno: boolean` from stable, it would be an extremely minimal change due to the lack of property conflict. Just add the following in `lib.deno.unstable.d.ts`:
```ts
declare interface WorkerOptions {
  deno?: boolean | {
    namespace: boolean,
    permissions: { ... },
  }
}
```
Since it's already runtime unstable, I think zero user workflows would break...

Reply:
@nayeemrmn sounds good, let's do this

Reply:
Also, shouldn't the interface be exported?